### PR TITLE
Update Xcode version to 13a233

### PR DESCRIPTION
**Pull request by yusuf-goog**
Update Xcode version to 13a233 for mac builders.

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by yusuf-goog** [comment]
For #94078

File: .ci.yaml
Comment (godofredoc) [inline .ci.yaml:69]: OS will also need to get changed to 12.0.1

**Review by godofredoc — COMMENTED** [review]
LGTM but let's wait for the ongoing release to get completed.

**Comment by jmagman** [comment]
Why did https://ci.chromium.org/p/flutter/builders/try/Mac%20build_aar_module_test/13799 run on a Mac-10.15.7 machine? https://chromium-swarm.appspot.com/bot?id=build887-m9
I was expecting the Mac tests to instead fail because there were no available staging machines that matched the requested OS.

**Comment by godofredoc** [comment]
> Why did https://ci.chromium.org/p/flutter/builders/try/Mac%20build_aar_module_test/13799 run on a Mac-10.15.7 machine? https://chromium-swarm.appspot.com/bot?id=build887-m9 I was expecting the Mac tests to instead fail because there were no available staging machines that matched the requested OS.

That's a devicelab test that we run on presubmit, .ci.yaml also needs an update for this test.

**Comment by CaseyHillers** [comment]
FYI tests are failing on:

```
xcodebuild: error: unable execute utility "/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/usr/bin/xcodebuild" because it requires a newer version of macOS.
```

**Comment by yusuf-goog** [comment]
Is there a good reason to peg it to specifically Mac-12.2? Should we set it to Mac-12?

**Comment by jmagman** [comment]
> Is there a good reason to peg it to specifically Mac-12.2? Should we set it to Mac-12?

I don't have a preference, this `mac_ios` combination is working:
https://github.com/flutter/flutter/blob/0ce527eb91a37ef31b60ed6d4a9613310f2107d4/.ci.yaml#L109-L111

File: .ci.yaml
Comment (yusuf-goog) [reply .ci.yaml:69]: Per @jmagman I changed all the mac builders to be at Mac-12, rather then pegged at a more specific version.

**Comment by jmagman** [comment]
> > Why did https://ci.chromium.org/p/flutter/builders/try/Mac%20build_aar_module_test/13799 run on a Mac-10.15.7 machine? https://chromium-swarm.appspot.com/bot?id=build887-m9 I was expecting the Mac tests to instead fail because there were no available staging machines that matched the requested OS.
> 
> That's a devicelab test that we run on presubmit, .ci.yaml also needs an update for this test.

Does that mean it will work post-merge once .ci.yaml config changes land?  This failure, for example, isn't in the devicelab: https://ci.chromium.org/p/flutter/builders/try/Mac%20build_tests_2_4/15803

**Comment by godofredoc** [comment]
> > > Why did https://ci.chromium.org/p/flutter/builders/try/Mac%20build_aar_module_test/13799 run on a Mac-10.15.7 machine? https://chromium-swarm.appspot.com/bot?id=build887-m9 I was expecting the Mac tests to instead fail because there were no available staging machines that matched the requested OS.
> > 
> > 
> > That's a devicelab test that we run on presubmit, .ci.yaml also needs an update for this test.
> 
> Does that mean it will work post-merge once .ci.yaml config changes land? This failure, for example, isn't in the devicelab: https://ci.chromium.org/p/flutter/builders/try/Mac%20build_tests_2_4/15803

Mac_build tests are hostonly tests, they use different .ci.yaml configurations from devicelab tests. All the tests need to pass on presubmit and as long as we keep half of the capacity running on 10.15  we should not break postsubmit tests as after landing the PR, postsubmit tests will still use the old configs for ~45mins until they are propagated. Once the configs are fully propagated it will be safe to upgrade the remaining machines.

**Comment by keyonghan** [comment]
New version of cocoon has been deploy with https://github.com/flutter/cocoon/pull/1597.
Manually triggered all the tests, and 12.2 is being picked up.

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/98210 override should also be reverted on this change.

**Comment by jmagman** [comment]
> #98210 override should also be reverted on this change.

Though I can to it in a follow-up.

**Comment by godofredoc** [comment]
> once .ci.yaml config changes land? This failure, for example, isn't in th
> Does that mean it will work post-merge once .ci.yaml config changes land? This failure, for example, isn't in the devicelab: https://ci.chromium.org/p/flutter/builders/try/Mac%20build_tests_2_4/15803

Yes, unfortunately .ci.yaml changes need to get propagated which takes ~45mins after landing the PR

**Comment by yusuf-goog** [comment]
mac tests are now passing with @jmagman latest fix.

**Review by godofredoc — APPROVED** [review]

**Comment by jmagman** [comment]
Not sure why Google testing hasn't updated, `Presubmit Test Passed`
Screen Shot 2022-02-10 at 5 04 51 PM

**Comment by godofredoc** [comment]
> Not sure why Google testing hasn't updated, `Presubmit Test Passed` Screen Shot 2022-02-10 at 5 04 51 PM

Seems like everything is ready to get landed. @yusuf-goog I'll be around if you need to check something after landing the PR

**Review by jmagman — APPROVED** [review]

**Comment by zanderso** [comment]
It looks like this is breaking the Mac web_tool_tests on the framework tree. Revert is here https://github.com/flutter/flutter/pull/98263.

**Pull request merged**
